quic-api.md

```diff
@@ -0,0 +1,284 @@
+
+Another particularity of QUIC stream is ability to explicitly close the writing side in the middle of work with the stream, see [`CompleteWrites`](https://learn.microsoft.com/en-us/dotnet/api/system.net.quic.quicstream.completewrites?view=net-7.0) or [`WriteAsync`](https://learn.microsoft.com/en-us/dotnet/api/system.net.quic.quicstream.writeasync?view=net-7.0#system-net-quic-quicstream-writeasync(system-readonlymemory((system-byte))-system-boolean-system-threading-cancellationtoken)) overload with `completeWrites` argument. The closing of the writing side lets the peer know that no more data will arrive but the peer still can continue sending (in case of a bidirectional stream). This is useful in scenarios like HTTP request/response exchange when the client sends the request and closes the writing side to let the server know that this is the end of the request content. Server is still able to send the response after that but knows that no more data will arrive from the client. And for erroneous cases, either writing or reading side of the stream can be aborted, see [`Abort`](https://learn.microsoft.com/en-us/dotnet/api/system.net.quic.quicstream.abort?view=net-7.0). The behavior of the individual methods for each stream type is summarized in the following table:
+
+|            | client  | server  |
+| -          | -       | -       |
+| `CanRead`  | _bidirectional_: `true`<br/> _unidirectional_: `false` | `true`  |
+| `CanWrite` | `true`  | _bidirectional_: `true`<br/> _unidirectional_: `false` |
+| `ReadAsync` | _bidirectional_: reads data<br/> _unidirectional_: `InvalidOperationException` | reads data |
+| `WriteAsync` | sends data => server read returns the data | _bidirectional_: sends data => client read returns the data<br/> _unidirectional_: `InvalidOperationException`  |
+| `CompleteWrites` | closes writing side => server read returns 0 | _bidirectional_: closes writing side => client read returns 0<br/> _unidirectional_: - |
+| `Abort(QuicAbortDirection.Read)` | _bidirectional_: [STOP_SENDING](https://www.rfc-editor.org/rfc/rfc9000.html#name-stop_sending-frames) => server write throws `QuicException(QuicError.OperationAborted)`<br/> _unidirectional_: - | [STOP_SENDING](https://www.rfc-editor.org/rfc/rfc9000.html#name-stop_sending-frames) => client write throws `QuicException(QuicError.OperationAborted)`|
+| `Abort(QuicAbortDirection.Write)` | [RESET_STREAM](https://www.rfc-editor.org/rfc/rfc9000.html#name-reset_stream-frames) => server read throws `QuicException(QuicError.OperationAborted)` | _bidirectional_: [RESET_STREAM](https://www.rfc-editor.org/rfc/rfc9000.html#name-reset_stream-frames) => client read throws `QuicException(QuicError.OperationAborted)`<br/> _unidirectional_: - |
```

Comment:
just my thoughts, would it be more clear, if instead of "-" in

> _unidirectional_: -

we would put something like "no-op"?

Reply:
Done.